### PR TITLE
feat(firebase_dynamic_links): `matchType` for pending Dynamic Link data for `iOS`.

### PR DESCRIPTION
**Pull request by russellwheatley**
## Description

See title.

## Related Issues

fixes https://github.com/firebase/flutterfire/issues/8462

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/match_type.dart [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  ///  The match has not been achieved.
+  none,
+
+  /// The match between the Dynamic Link and this device may not be perfect, hence you should
+  /// not reveal any personal information related to the Dynamic Link.
+  weak,
+
+  /// The match between the Dynamic Link and this device has high confidence but small possibility
+  /// of error still exist.
+  defaultMatchType,
```
Comment (Salakar) [inline packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/match_type.dart:17]: Not sure I'm a fan in the variations of the naming convention for this one - but I'm also not sure what a better alternative could be when `default` is a keyword in Dart 🤷

Comment (russellwheatley) [reply packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/match_type.dart:17]: @Salakar Yeah, exactly. Will change to whatever is best but we can't have `default` unfortunately.

**Review by Salakar — APPROVED** [review]

File: packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/match_type.dart [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  ///  The match has not been achieved.
+  none,
+
+  /// The match between the Dynamic Link and this device may not be perfect, hence you should
+  /// not reveal any personal information related to the Dynamic Link.
+  weak,
+
+  /// The match between the Dynamic Link and this device has high confidence but small possibility
+  /// of error still exist.
+  highMatchType,
```
Comment (Salakar) [inline packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/match_type.dart:17]:
```suggestion
  high,
```

File: packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/method_channel/utils/convert_match_type.dart [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import '../../match_type.dart';
+
+MatchType? convertMatchType(int? matchType) {
+  switch (matchType) {
+    case 0:
+      return MatchType.none;
+    case 1:
+      return MatchType.weak;
+    case 2:
+      return MatchType.highMatchType;
```
Comment (Salakar) [inline packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/method_channel/utils/convert_match_type.dart:14]:
```suggestion
      return MatchType.high;
```

File: packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/test/pending_dynamic_link_data_test.dart [hunk 4]
```diff
@@ -8,14 +8,17 @@ import 'package:flutter_test/flutter_test.dart';
 void main() {
   Uri link = Uri.parse('pending-link');
   int minimumVersion = 12;
+  MatchType matchType = MatchType.highMatchType;
```
Comment (Salakar) [inline packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/test/pending_dynamic_link_data_test.dart:11]:
```suggestion
  MatchType matchType = MatchType.high;
```

File: packages/firebase_dynamic_links/firebase_dynamic_links/test/firebase_dynamic_links_test.dart [hunk 5]
```diff
@@ -88,6 +88,7 @@ void main() {
         const mockClickTimestamp = 1234567;
         const mockMinimumVersionAndroid = 12;
         const mockMinimumVersionIOS = 'ios minimum version';
+        const mockMatchTypeIOS = MatchType.highMatchType;
```
Comment (Salakar) [inline packages/firebase_dynamic_links/firebase_dynamic_links/test/firebase_dynamic_links_test.dart:91]:
```suggestion
        const mockMatchTypeIOS = MatchType.high;
```

**Review by Salakar — APPROVED** [review]

**Review by dackers86 — APPROVED** [review]

**Pull request merged**